### PR TITLE
Add `contributors` field

### PR DESCRIPTION
**Pull request by Zack-83**
**Related issues**

Refs: #66 #84 

(For autoclosure of issues when PR is merged use `Fixes #<issue-number>` syntax)

**Describe the changes made in this pull request**

This PR adds an additional field `contributors` as an alternative to `authors` but with a secundary role, coherently with DataCite and other metadata schemas commonly used to document data/software publications. 

**Review checklist**

- [x] Please check if the pull request is against the correct branch  
(format/schema/semantic documentation changes: `develop`; typos, meta files, etc.: `main`)
- [x] Please check if all changes are recorded in `CHANGELOG.md` and adapt if necessary.
- [ ] Please run tests locally.

```bash
cd $(mktemp -d --tmpdir cff.XXXXXX)
git clone https://github.com/citation-file-format/citation-file-format .
git checkout <branch>
python3 -m venv env
source env/bin/activate
pip install --upgrade pip wheel setuptools
pip install -r requirements.txt
pytest
<do other things>
```

- [ ] `<do other things>`

### CODE REVIEW

**Comment by dmoracze** [comment]
Alternative solution that maps onto an already existing taxonomy:

https://github.com/citation-file-format/citation-file-format/issues/112#issuecomment-1610077526

**Comment by sdruskat** [comment]
Thanks, @Zack-83, for this pull request!

We'd be looking to back this up further with a vocabulary of roles, based on an (ideally community-approved) taxonomy (see https://github.com/citation-file-format/citation-file-format/issues/112#issuecomment-1610851858). While contribution roles are going to be part of version 1.3.0, vetting suitable taxonomies is still ongoing. I'm happy to look into basing this work on your PR, but it will likely stay open for quite some time.

/cc @jspaaks

**Comment by jspaaks** [comment]
@sdruskat how would you feel about merging this into `develop` as-is, and then making changes on top of that as we reach consensus of where we want to go taxonomy-wise?

File: schema.json [hunk 1]
```diff
@@ -1775,6 +1775,22 @@
             "type": "array",
             "uniqueItems": true
         },
+        "contributors": {
+            "description": "Persons or entities who played a secundary role to the generation of the software or dataset.",
```
Comment (sdruskat) [inline schema.json:1779]:
```suggestion
            "description": "The contributor(s) to a work.",
```

**suggestion:** Word this in analogy with `authors`.

**Review by sdruskat — COMMENTED** [review]
Thanks for the PR!

I agree with @jspaaks after all. I've made one small suggestion to change description wording. Other than that it's goo to merge once the CI pipelines have run and passed.

File: schema.json (hunk 1 above)
Comment (Zack-83) [reply schema.json:1779]: Fine for me, if the others agree.

**Comment by Zack-83** [comment]
> Hi @Zack-83, can you please pull in the latest changes in `develop`? This should fix the failing tests.
> 
> Something like:
> 
> ```shell
> git merge upstream/develop
> ```
> 
> Thanks!

Sorry, it seems I cannot. Maybe I'm just too stupid.

In the Git language, `origin` is my fork and `upstream` is the original repository, or the opposite?

grafik

grafik

File: schema.json (hunk 1 above)
Comment (jspaaks) [reply schema.json:1779]: OK with me!

**Comment by jspaaks** [comment]
No worries, you're not stupid :) git is nice but tricky at times

`upstream` is https://github.com/citation-file-format/citation-file-format.git, `origin` is your fork. You can use different names but these are conventional. If you want to be able to use both at the command line you can add a remote:

```shell
# add "upstream"
git remote add upstream https://github.com/citation-file-format/citation-file-format.git

# check what you have:
git remote -v

git fetch upstream
git merge upstream/develop
```

or maybe a simpler approach is to go to your fork on github, click the Sync fork button

image

**Comment by jspaaks** [comment]
Sorry my bad, please pull in latest changes from `origin/main` into your `develop` as well

**Comment by jspaaks** [comment]
One way is to 
1. sync your main with upstream main using the button
1. in a terminal, cd to your repo
1. `git checkout develop`
1. `git fetch origin`
1. `git merge origin/main`
1. `git push origin develop`

that should do it

**Comment by Zack-83** [comment]
Hey people,
i only did it worse. Before were 2 changed files, now 14. 41 commits to stage.
Probably my mistake - i rebased onto `upstream/main`.
It has become late and I wil be off the next two weeks. Might one of you be so kind and solve the problem? I invited you both as collaborators.
After the 24th of July I am also reachable on my office telephone.
Thanks,
G.

**Review by jspaaks — APPROVED** [review]
Great stuff, thanks for sticking with us :)

**Pull request merged**